### PR TITLE
ISC - Restructure - Learn and Build

### PR DESCRIPTION
**Pull request by lucas-tortora**
# Description of change

Build and learn explanations restructure. Alternative to https://github.com/iotaledger/iota-wiki/pull/1453

## Links to any relevant issues

fixes https://github.com/iotaledger/devx/issues/314
fixes https://github.com/iotaledger/devx/issues/244
fixes https://github.com/iotaledger/devx/issues/266
fixes https://github.com/iotaledger/devx/issues/317

## Type of change

- Documentation 

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the [contribution guidelines](https://github.com/iota-wiki/iota-wiki/blob/main/.github/CONTRIBUTING.md) for this project
- [x] I have performed a self-review of my own changes
- [x] I have made sure that added/changed links still work
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by lucas-tortora** [comment]
https://iota-wiki-jjwtgu483-iota1.vercel.app/

**Comment by lucas-tortora** [comment]
Preview: https://iota-wiki-6mn4lyyyj-iota1.vercel.app/

File: docs/learn/smart-contracts/smart-contract-execution.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+- Validators
+- User Requests
+- Gas Fees
+---
+
+# Smart Contract Execution
+
+## TL;DR;
+
+- [Introduction](#introduction): Smart contracts are self-executing contracts on blockchain networks, activated by validators in response to user requests.
```
Comment (jlvandenhout) [inline docs/learn/smart-contracts/smart-contract-execution.md:16]: I'm not sure I understand what is meant with "self-executing contracts". I'd prefer to just call them programs, ran by validators upon request.

File: docs/learn/smart-contracts/protocol-overview.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+## Smart Contracts
+
+Smart contracts are like digital programs on a blockchain, consisting of functions that either view or update the
+blockchain's state based on given inputs. They are versatile, written in languages like Solidity, Rust, or Go, and are
+compiled into bytecode. This bytecode is then executed by a [virtual machine](#virtual-machines), acting as the
+blockchain's engine to process these contracts.
+
+## Virtual Machines
+
+[Validators](#validators) run th virtual machines are the powerhouses that run smart contracts. They take the
```
Comment (jlvandenhout) [inline docs/learn/smart-contracts/protocol-overview.md:36]:
```suggestion
Virtual machines, operated by [Validators](#validators), are the execution engines of smart contracts that understand smart contract bytecode and are able compute state changes. They take the
```

File: docs/learn/smart-contracts/protocol-overview.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+Validators are critical components in a blockchain network. They operate the virtual machines and are responsible for
+storing the [state](accounts-addresses-and-fees.md) and history of the blockchain. Validators process requests to run
+smart contracts,
+achieve consensus on state updates, and create a blockchain that represents the history of said state updates.
+
+## State
+
+The state in a [blockchain](#blockchains) is the data that smart contracts interact with and modify. It's a record of
+all changes made by executing smart contracts, committed in blocks, and forms the backbone of the blockchain's history.
```
Comment (jlvandenhout) [inline docs/learn/smart-contracts/protocol-overview.md:54]:
```suggestion
The state in a [blockchain](#blockchains) is a key-value store that smart contracts interact with and modify. It's a result of smart contract execution, committed in blocks, forming a blockchain representing the history of state updates leading to the current state.
```

**Review by jlvandenhout — CHANGES_REQUESTED** [review]

File: docs/learn/smart-contracts/smart-contract-execution.md (hunk 1 above)
Comment (lucas-tortora) [reply docs/learn/smart-contracts/smart-contract-execution.md:16]:
```suggestion
- [Introduction](#introduction): Smart contracts are programs run by validators upon request by validators in response to user requests.
```

**Comment by lucas-tortora** [comment]
https://iota-wiki-oc20mbzfd-iota1.vercel.app/

**Review by Dr-Electron — APPROVED** [review]

File: docs/learn/smart-contracts/smart-contract-execution.md (hunk 1 above)
Comment (Dr-Electron) [reply docs/learn/smart-contracts/smart-contract-execution.md]: Why is this resolved?

File: docs/learn/smart-contracts/smart-contract-execution.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+- [Structure of a Smart Contract](#structure-of-a-smart-contract): Comprising functions within a decentralized framework, smart contracts have a unique state and accounts, managed by validator node consensus.
+- [View Entry Points](#view-entry-points): Read-only operations in smart contracts that do not modify data or incur execution fees.
+- [Full Entry Points](#full-entry-points): Functions that can alter a contract's state and typically require execution fees.
+- [Invoking a Smart Contract](#invoking-a-smart-contract): Activation of smart contracts through requests, which can be off-ledger or on-ledger, with associated gas fees for computational resources.
+
+## Introduction
+
+Smart contracts are essentially self-executing contracts, run on blockchain networks. Network nodes, also known as
```
Comment (Dr-Electron) [inline docs/learn/smart-contracts/smart-contract-execution.md:24]: Same change needed as before I guess

**Review by Dr-Electron — CHANGES_REQUESTED** [review]

File: docs/learn/smart-contracts/smart-contract-execution.md (hunk 4 above)
Comment (lucas-tortora) [reply docs/learn/smart-contracts/smart-contract-execution.md:24]:
```suggestion
Smart contracts are self-executing contracts run on blockchain networks. Network nodes, also known as 
```

File: docs/learn/smart-contracts/smart-contract-execution.md (hunk 1 above)
Comment (lucas-tortora) [reply docs/learn/smart-contracts/smart-contract-execution.md]: don't know, it seems the suggestion didn't go through but it still resolved...weird. Anyhow, updated

File: docs/learn/smart-contracts/smart-contract-execution.md [hunk 5]
```diff
@@ -13,15 +13,15 @@ keywords:
...
+- [Introduction](#introduction): Smart contracts are programs run by [validators](chain-owners-validators-and-access-nodes.md#validator-nodes) upon requests.
 - [Structure of a Smart Contract](#structure-of-a-smart-contract): Comprising functions within a decentralized framework, smart contracts have a unique state and accounts, managed by validator node consensus.
 - [View Entry Points](#view-entry-points): Read-only operations in smart contracts that do not modify data or incur execution fees.
 - [Full Entry Points](#full-entry-points): Functions that can alter a contract's state and typically require execution fees.
 - [Invoking a Smart Contract](#invoking-a-smart-contract): Activation of smart contracts through requests, which can be off-ledger or on-ledger, with associated gas fees for computational resources.
 
 ## Introduction
 
-Smart contracts are essentially self-executing contracts, run on blockchain networks. Network nodes, also known as 
+Smart contracts are essentially programs run by upon requests. Network nodes, also known as
```
Comment (jlvandenhout) [inline docs/learn/smart-contracts/smart-contract-execution.md:24]: I guess you meant to say something like this?

```suggestion
Smart contracts are essentially programs run by a network of nodes upon requests. Network nodes, also known as 
```

**Comment by lucas-tortora** [comment]
Updated preview: https://iota-wiki-gr634oug3-iota1.vercel.app/

File: docs/maintain/wasp/v1.0.0-rc.6/sidebars.js [hunk 6]
```diff
@@ -16,16 +16,45 @@ module.exports = {
   // But you can create a sidebar manually
   tutorialSidebar: [
```
Comment (Dr-Electron) [inline docs/maintain/wasp/v1.0.0-rc.6/sidebars.js:17]: Can we finally not use this stupid name anymore :trollface:

File: versionedConfig.js [hunk 7]
```diff
@@ -22,35 +22,9 @@ exports.buildPluginsConfig = [
...
-      },
-    ],
-  },
-  {
-    id: 'wasp-wasm',
-    label: 'WASM (Experimental)',
-    description: 'Create WASM smart contracts.',
+    id: 'isc',
+    label: 'Smart contracts',
+    description: 'Create smart contracts.',
```
Comment (Dr-Electron) [inline versionedConfig.js:27]:
```suggestion
    label: 'Smart Contracts',
    description: 'Create Smart Contracts.',
```

**Review by Dr-Electron — CHANGES_REQUESTED** [review]

File: docs/build/isc/v1.0.0-rc.6/docs/getting-started/languages-and-vms.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+Every deployed IOTA Smart Contracts chain automatically includes a core contract
+called [`evm`](../reference/core-contracts/evm.md). This core contract is responsible for running EVM code and
+storing the EVM state.
+
+The Wasp node also provides a standard JSON-RPC service, which allows you to interact with the EVM layer using existing
+tooling like [MetaMask](https://metamask.io/), [Remix](https://remix.ethereum.org/) or [Hardhat](https://hardhat.org/).
+Deploying EVM contracts is as easy as pointing your tools to the JSON-RPC endpoint.
+
+:::warning EVM Compatibility
```
[the remark this answers is not in the document]
Comment (lucas-tortora) [reply docs/build/isc/v1.0.0-rc.6/docs/getting-started/languages-and-vms.md:41]: so far we use it once, but then again it doesn't hurt to do it

File: docs/maintain/wasp/v1.0.0-rc.6/sidebars.js (hunk 6 above)
Comment (Dr-Electron) [reply docs/maintain/wasp/v1.0.0-rc.6/sidebars.js:17]:
```suggestion
  sidebar: [
```

File: docs/build/isc/v1.0.0-rc.6/docs/getting-started/compatibility.md [hunk 9]
```diff
@@ -31,7 +31,9 @@ Ethereum tools think they are interfacing with an actual Ethereum node, but some
 
 :::warning
 
-There is a difference in the decimal precision of ether (18 decimal places) to MIOTA/SMR(6 decimal places). Because of this, when sending native tokens in the EVM, which are expressed in wei (ether = 10<sup>18</sup>wei), the last 12 decimal places will be ignored.
+There is a difference in the decimal precision of ether (18 decimal places) to MIOTA/SMR(6 decimal places).
```
[the remark this answers is not in the document]
Comment (lucas-tortora) [reply docs/build/isc/v1.0.0-rc.6/docs/getting-started/compatibility.md:34]: Yup, I read the team highlights :)

**Comment by lucas-tortora** [comment]
https://iota-wiki-2kuf9xxjt-iota1.vercel.app/

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/EVM/send-funds-from-L1-to-L2.md [hunk 10]
```diff
@@ -0,0 +1,105 @@
+---
+description: How to fund your L2 account.
```
Comment (Dr-Electron) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/EVM/send-funds-from-L1-to-L2.md:2]: Wrong description and wrong sidebar title 🤔

File: docs/build/isc/v1.0.0-rc.6/docs/getting-started/tools.mdx
Comment (Dr-Electron) [inline docs/build/isc/v1.0.0-rc.6/docs/getting-started/tools.mdx:1]: Looks good. But I would change the order of tools. Move wasp-cli down for example. Most people won't need it

File: docs/build/isc/v1.0.0-rc.6/docs/schema/how-tos/state.mdx [hunk 11]
```diff
@@ -173,7 +173,7 @@ export class MutableDividendState extends wasmtypes.ScProxy {
 
 Examining the `state.xx`code, generated by the [Schema Tool](usage.mdx),
 you can find the `MutableDividendState` struct.
-This interface allows type-safe access to each state variable through mutable [proxies](../../explanations/proxies.mdx),
+This interface allows type-safe access to each state variable through mutable [proxies](/isc/explanations/proxies),
```
Comment (Dr-Electron) [inline docs/build/isc/v1.0.0-rc.6/docs/schema/how-tos/state.mdx:176]: Why is this still not a file path?

**Review by Dr-Electron — APPROVED** [review]
Small things but other than that, lgtm

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/EVM/send-funds-from-L1-to-L2.md (hunk 10 above)
Comment (Dr-Electron) [reply docs/build/isc/v1.0.0-rc.6/docs/how-tos/EVM/send-funds-from-L1-to-L2.md]: Fixed in https://github.com/iotaledger/iota-wiki/pull/1415/commits/70576612a9366b177f4ff2dff9685094cc1cb0cc

File: docs/build/isc/v1.0.0-rc.6/docs/getting-started/tools.mdx
Comment (Dr-Electron) [reply docs/build/isc/v1.0.0-rc.6/docs/getting-started/tools.mdx:1]: https://github.com/iotaledger/iota-wiki/pull/1415/commits/14ce868ea64431d113fc307ac0c0ed0a8f738ce5

File: docs/build/isc/v1.0.0-rc.6/docs/schema/how-tos/state.mdx (hunk 11 above)
Comment (Dr-Electron) [reply docs/build/isc/v1.0.0-rc.6/docs/schema/how-tos/state.mdx]: https://github.com/iotaledger/iota-wiki/pull/1415/commits/67b4cecd33c24aa1152ce413f4cb550bf0ae4dc5

**Review by jlvandenhout — APPROVED** [review]

**Pull request merged**